content/en/docs/concepts/cluster-administration/system-logs.md

```diff
@@ -231,6 +231,53 @@ Similar to the container logs, you should rotate system component logs in the `/
 In Kubernetes clusters created by the `kube-up.sh` script, log rotation is configured by the `logrotate` tool.
 The `logrotate` tool rotates logs daily, or once the log size is greater than 100MB.
 
+## Log query
+
+{{< feature-state for_k8s_version="v1.27" state="alpha" >}}
+
+To help with debugging issues on nodes, Kubernetes v1.27 introduced a feature that allows viewing logs of services
+running on the node. To use the feature, ensure that the `NodeLogQuery`
+[feature gate](/docs/reference/command-line-tools-reference/feature-gates/) is enabled for that node, and that the
+kubelet configuration options `enableSystemLogHandler` and `enableSystemLogQuery` are both set to true. On Linux
+we assume that service logs are available either via journald. On Windows we assume that service logs are available
+in the application log provider. On both operating systems, logs are also available by reading files within
+`/var/log/`.
+
+A node level administrator can try out this alpha feature on all their nodes, or on just a subset. Provided you're
+authorized to do so, you can now query logs for system services on that node:
+```shell
+# Fetch kubelet logs from a node named node-1.example
+kubectl get --raw "/api/v1/nodes/node-1.example/proxy/logs/?query=kubelet"
+```
+
+You can also fetch files, provided that the files are in a directory that the kubelet allows for log
+fetches. For example, you can fetch a log from `/var/log` on a Linux node:
+```shell
+kubectl get --raw "/api/v1/nodes/<insert-node-name-here>/proxy/logs/?query=/<insert-log-file-name-here>"
+```
```

Comment:
There must be some implicit restrictions to the `/<insert-log-file-name-here>`.
It sounds like I'm able to read the whole file system if the "node admin"
has not taken some special care of permissions.
By default `kubelet` runs as a root ... that is so scary to me.

Reply:
The capability to view the `/var/log` directory using the logs endpoint via the proxy was already present in the kubelet. All this feature is doing is extending it to also query the journal logs. As part of that we also want to retain the existing functionality to view the log files in `/var/log` using the `--query` option. 

I also want to call out that this feature was discussed in the [sig-arch call on Feb 9th, 2023](https://docs.google.com/document/d/1BlmHq5uPyBUDlppYqAAzslVbAO8hilgjqZUTaNXUhKM/edit#). The implementation was an outcome of that call.

Reply:
To be clear, `kubectl get --raw "/api/v1/nodes/<insert-node-name-here>/proxy/logs/<insert-log-file-name-here>` will work even without this feature.